### ISSUE TITLE
indefinite retry with output modules doesn't work

### ISSUE THREAD

**Issue by Kane-Sendgrid** [rsyslog/rsyslog#808]
I'm trying to configure indefinite retries with dummy output module. I've modified doAction and endTransaction to always return RS_RET_SUSPENDED. Configuration is as following:

`queue.spoolDirectory="/var/spool/rsyslog" queue.filename="some_filename" queue.type="Disk" queue.maxdiskspace="1G" queue.saveonshutdown="on" action.resumeRetryCount="-1" action.reportSuspensionContinuation="on" action.resumeInterval="1"`

After sending test line to logger:

```
6130.843876518:action 0 queue:Reg/w0: Action 0 transitioned to state: rdy
6130.844112940:action 0 queue:Reg/w0: Action 0 transitioned to state: itx
6130.844131796:action 0 queue:Reg/w0: Action 0 transitioned to state: rtry
6130.844137106:action 0 queue:Reg/w0: actionCommit, in retry loop, iRet -2007
6130.844142731:action 0 queue:Reg/w0: regular consumer finished, iret=-2007, szlog 0 sz phys 1
6130.844149164:action 0 queue:Reg/w0: DeleteProcessedBatch: we deleted 1 objects and enqueued 0 objects
6130.844154698:action 0 queue:Reg/w0: doDeleteBatch: delete batch from store, new sizes: log 0, phys 0
6130.844160252:action 0 queue:Reg/w0: regular consumer finished, iret=4, szlog 0 sz phys 0
6130.844165414:action 0 queue:Reg/w0: action 0 queue:Reg/w0: worker IDLE, waiting for work.
```

And then it stays in this state without retrying the message.

**Comment by rgerhards** [comment]
Note:  I need to find a way to make it fail with an action that I can test easily, like omfwd (I don't have
kafka know how and so I do not know how to craft a consistent test case for it). If that fails, I'll write a testing plugin just for that purpose (or extend omtesting, to be seen).

**Comment by Kane-Sendgrid** [comment]
Reproduced with omrelp with following configuration:
```module(load="omrelp")
local0.* action(type="omrelp"
    target="10.0.0.1"
    port="2514"
    name="debug_relp"
    queue.filename="relp_queue"
    queue.maxdiskspace="1G"
    queue.saveonshutdown="on"
    queue.spoolDirectory="/var/spool/rsyslog"
    queue.type="Disk"
    action.resumeRetryCount="-1"
    action.resumeInterval="1"
)```

**Comment by rgerhards** [comment]
This is very, very probably an incarnation of https://github.com/rsyslog/rsyslog/issues/974, which I am currently working on.

**Comment by rgerhards** [comment]
Basic problem, and especially the one reported here specifically, was already fixed in 8.20.0, but I forgot to close this tracker here. I kept #974 open as I want to further improve on the handling, but for the tracker here, everything should actually be done.
